sympy/stats/rv_interface.py

```diff
@@ -148,6 +152,10 @@ def covariance(X, Y, condition=None, **kwargs):
     >>> covariance(X, Y + rate*X)
     1/lambda
     """
+    if pspace(X) == PSpace() or pspace(Y) == PSpace():
```

Comment:
So, you are checking if the pspace is independent or not?

Reply:
Here, dummy random variables, which don't have Probability space are mapped to `PSpace()` which does not have methods compute `E`, `P`, `variance`, and `covariance`, are kept in view. So, for them, we are returning objects.
https://github.com/sympy/sympy/blob/5d1b2c6f04f583ec0beeea4b8238d4151b497332/sympy/stats/rv.py#L241-L252